hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/schemas/InitialModServiceTokenSchema.java

```diff
@@ -184,8 +190,47 @@ public void setStakingFs(
         this.mnc = mnc;
     }
 
+    /**
+     * Updates in-state staking info to match the address book.
+     * <ol>
+     *     <li>For any node with staking info in state that is no longer in the address book,
+     *     marks it deleted and sets its weight to zero.</li>
+     *     <li>For any node in the address book that is not in state, initializes its staking info.</li>
+     * </ol>
+     *
+     * @param ctx {@link MigrationContext} for this schema restart operation
+     */
     @Override
-    public void restart(@NonNull MigrationContext ctx) {}
+    public void restart(@NonNull MigrationContext ctx) {
+        final var networkInfo = ctx.networkInfo();
+        final var newStakingInfos = ctx.newStates().<EntityNumber, StakingNodeInfo>get(STAKING_INFO_KEY);
```

Comment:
We should work with the store here, IMO, and not with `WritableKVState`.

Reply:
Fixed.